### PR TITLE
Make tests passing on Python 3.7

### LINKED ISSUE #3102: Fix (make robust) test_timer_context_cancelled under Python 3.7
## Long story short

With emerging of 3.7, we need to adjust test suite to pass under this env

## Expected behaviour

test passes

## Actual behaviour

One of coroutine cancellation tests fails:

```python-traceback
$ make cov-ci-no-ext
...
――――――――――――――――――――――――― test_timer_context_cancelled ―――――――――――――――――――――――――
    def test_timer_context_cancelled():
        with mock.patch('aiohttp.helpers.asyncio') as m_asyncio:
            m_asyncio.TimeoutError = asyncio.TimeoutError
            loop = mock.Mock()
            ctx = helpers.TimerContext(loop)
            ctx.timeout()
    
            with pytest.raises(asyncio.TimeoutError):
                with ctx:
                    pass
    
>           assert m_asyncio.Task.current_task.return_value.cancel.called
E           AssertionError: assert False
E            +  where False = <MagicMock name='asyncio.Task.current_task().cancel' id='23073207868104'>.called
E            +    where <MagicMock name='asyncio.Task.current_task().cancel' id='23073207868104'> = <MagicMock name='asyncio.Task.current_task()' id='23073207876128'>.cancel
E            +      where <MagicMock name='asyncio.Task.current_task()' id='23073207876128'> = <MagicMock name='asyncio.Task.current_task' id='23073207863616'>.return_value
E            +        where <MagicMock name='asyncio.Task.current_task' id='23073207863616'> = <MagicMock name='asyncio.Task' id='23073207799992'>.current_task
E            +          where <MagicMock name='asyncio.Task' id='23073207799992'> = <MagicMock name='asyncio' id='23073210485560'>.Task
tests/test_helpers.py:430: AssertionError
```

## Steps to reproduce

See https://travis-ci.com/aio-libs/aiohttp/jobs/130964628#L793-L814

## Your environment

Travis CI

### PR DESCRIPTION
**Pull request by asvetlov**
Superseeds #3113
Fixes #3102

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -135,7 +135,9 @@ os: linux
 jobs:
   fast_finish: true
   allow_failures:
-  - python: nightly
+  - python:
+    - 3.7-dev
```
Comment (webknjaz) [inline .travis.yml]: it doesn't work like this. you should have separate entries in `allow_failures`

Comment (asvetlov) [reply .travis.yml]: Aha, done

**Comment by asvetlov** [comment]
I've added dist parameter but it doesn't help.
Removed it as unrelated to the PR.

**Comment by asvetlov** [comment]
Since Python 3.7 is not supported by travis yet let's merge the PR with allowed failure on 3.7-dev.
3.7 release is a little different from 3.7-dev on travis.
I have all tests passed on Python 3.7.0 locally.

**Comment by webknjaz** [comment]
well, it's kinda supported. you just need to combine `dist: xenial` + `sudo: required` specifically for job with `3.7.0`

**Pull request merged**